src/App.tsx

```diff
@@ -15,48 +15,178 @@ export const goodsFromServer = [
   'Garlic',
 ];
 
+enum SortType {
+  NONE,
+  ALPHABET,
+  LENGTH,
+}
+
+type ReorderOptions = {
+  sortType: SortType;
+  isReversed: boolean;
+};
+
+export function getReorderedGoods(
+  goods: string[],
+  { sortType, isReversed }: ReorderOptions,
+) {
+  const visibleGoods = [...goods];
+
+  visibleGoods.sort((good1, good2) => {
+    switch (sortType) {
+      case SortType.ALPHABET:
+        return good1.localeCompare(good2);
+
+      case SortType.LENGTH:
+        return good1.length - good2.length;
+
+      default:
+        return 0;
+    }
+  });
+
+  if (isReversed) {
+    visibleGoods.reverse();
+  }
+
+  return visibleGoods;
+}
+
 export const App: React.FC = () => {
+  const [reorderOptions, setReorderOptions] = useState({
+    sortType: SortType.NONE,
+    isReversed: false,
+  });
+
+  const reoderedGoods = getReorderedGoods(goodsFromServer, reorderOptions);
+
+  const sortAlphabeticallyRef = useRef<any>(null);
+  const sortByLengthRef = useRef<any>(null);
```

Comment:
u don't need to use refs in this task at all. U can apply classnames based on the values from state